concert/ext/tangoservers/walker.py

```diff
@@ -182,40 +183,48 @@ async def write_sequence(self, name):
             self.set_state(DevState.FAULT)
             raise
 
-
-
-
     async def consume(self, walker: DirectoryWalker) -> None:
+        """Defines an internal consumer coroutine to write incoming stream"""
         with PerformanceTracker() as pt:
             total_bytes = await walker.write(self._receiver.subscribe())
             pt.size = total_bytes * q.B
 
     @DebugIt()
     @command(
-        dtype_in=str,
-        doc_in="file name for the log file"
+        dtype_in=(str,),
+        doc_in="identifier, log_name, log_level and file_name for logging"
     )
-    async def open_log_handler(self, file_name: str) -> None:
-        self._log_handler = logging.FileHandler(os.path.join(self._current, file_name))
-        self._logger.addHandler(self._log_handler)
+    async def register_logger_with(self, args: Tuple[str, str, str, str]) -> None:
```

Comment:
This should raise an error if I want to register the same UID twice.

Reply:
Also, why do I need UID? The file name is unique, cannot that server as an identifier?

Reply:
The uid to my thought is the distinguisher between Logger instances. When @MarcusZuber  mentioned that multiple Directors and Experiments could be nested, I understood it as, multiple instances of these individual classes may exist at the same time. If that understanding is correct then file name cannot be a distinguisher. It is not unique. Therefore, I chose to create a UID from the instance itself. When we get a string out of the instance it looks something like this: `<__main__.SomeClass object at 0x7f8291ef66a0>`. It has the starting memory location. It should be unique for an object.

Reply:
I would say, that the absolute filename should be unique enough. Otherwise, we can have multiple loggers writing to the same file, which is also maybe not the best idea.

The nested directors will create sub-folders, so each iteration will get a new file.
E.g.
```python 
exp = await MyExperiment()
energy_scan_director = await EnergyDirector(exp)
z_scan_director = await ZScanDirecto(energy_scan_director)
await z_scan_director.run()
```
will result in such a file structure:
- director.log (from z-scan)
- iteration_0_z
  - director.log  (from energy_scan)
  - iteration_0_energy
    - experiemnt.log
  - iteration_1_energy
    - experiment.log
- iteration_1_z
  - director.log (from energy_scan)
  - iteration_0_energy
    - experiemnt.log
  - iteration_1_energy
    - experiment.log